src/lib/factories.tsx

```diff
@@ -105,18 +105,16 @@ function createShorthandFromValue(
   const valIsNoop = _.isNil(value) || typeof value === 'boolean'
   if (valIsNoop && !options.render) return null
 
+  // return value as is if it a ReactElement
+  if (React.isValidElement(value)) {
+    return value as React.ReactElement<Props>
+  }
+
   const valIsPrimitive = typeof value === 'string' || typeof value === 'number'
   const valIsPropsObject = _.isPlainObject(value)
-  const valIsReactElement = React.isValidElement(value)
 
   // unhandled type warning
-  if (
-    process.env.NODE_ENV !== 'production' &&
-    !valIsPrimitive &&
-    !valIsPropsObject &&
-    !valIsReactElement &&
-    !valIsNoop
-  ) {
+  if (process.env.NODE_ENV !== 'production' && !valIsPrimitive && !valIsPropsObject && !valIsNoop) {
```

Comment:
suggest this check being left as it was before, and put all this section to the top of the method. 

This one has served as precondition invariant, and thus:
- it is aimed to clearly indicates method's expectations about input data (`value`)
  - currently this expectation condition has changed in code, while, in fact, method's precondition hasn't changed anyhow
- it should run before all the method's logic